test/orange/surf/RecursiveSimplifier.test.cc

```diff
@@ -0,0 +1,91 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2023 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file orange/surf/RecursiveSimplifier.test.cc
+//---------------------------------------------------------------------------//
+#include "orange/surf/RecursiveSimplifier.hh"
+
+#include <iomanip>
+#include <sstream>
+
+#include "orange/surf/SurfaceIO.hh"
+
+#include "celeritas_test.hh"
+
+namespace celeritas
+{
+namespace test
+{
+//---------------------------------------------------------------------------//
+
+class RecursiveSimplifierTest : public ::celeritas::test::Test
+{
+  protected:
+};
+
+TEST_F(RecursiveSimplifierTest, single)
+{
+    std::vector<std::string> result;
+
+    RecursiveSimplifier simplify{[&](Sense sense, auto&& surf) {
```

Comment:
In general this is bad practice, as it changes the type from `RecursiveSimplifier<{functor}>` to `RecursiveSimplifier<{functor}&>` (potentially more indirection and extending the lifetime+scope of the functor). Since this is a unit test, readability is a plus so I'll implement that change in this file.